src/common/tab/tabView.component.tsx

```diff
@@ -0,0 +1,98 @@
+import AttachmentOutlinedIcon from '@mui/icons-material/AttachmentOutlined';
+import CollectionsOutlinedIcon from '@mui/icons-material/CollectionsOutlined';
+import { Tabs } from '@mui/material';
+import Grid from '@mui/material/Grid';
+import React from 'react';
+import { useSearchParams } from 'react-router-dom';
+import { a11yProps, StyledTab } from '../../common/tab/tab.utils';
+import TabPanel from '../../common/tab/tabPanel.component';
+
+export interface TabViewProps<T> {
+  defaultTab: T;
+  ariaLabelPrefix: string;
+  tabData: {
+    value: T;
+    icon?: React.ReactElement;
+    component: React.ReactNode;
+  }[];
+  gallery?: boolean;
+  attachments?: boolean;
+}
+
+function TabView<T>(props: TabViewProps<T>) {
+  const { defaultTab, ariaLabelPrefix, tabData, gallery, attachments } = props;
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  // Retrieve the tab value from the URL or default to the passed defaultTab prop
+  const urlTabValue = (searchParams.get('tab') as T) || defaultTab;
+  const [tabValue, setTabValue] = React.useState<T>(urlTabValue);
+
+  const handleTabChange = React.useCallback(
+    (_event: React.SyntheticEvent, newValue: T) => {
+      setTabValue(newValue);
+      if (newValue === defaultTab) {
+        searchParams.delete('tab');
+        setSearchParams(searchParams, { replace: true });
+      } else {
+        setSearchParams(
+          { tab: newValue as unknown as string },
+          { replace: true }
+        );
+      }
+    },
+    [defaultTab, searchParams, setSearchParams]
+  );
+
+  // Optionally add Gallery and Attachments tabs
+  if (gallery) {
+    tabData.push({
+      value: 'Gallery' as T,
+      icon: <CollectionsOutlinedIcon />,
+      component: <></>,
+    });
+  }
+
+  if (attachments) {
+    tabData.push({
+      value: 'Attachments' as T,
+      icon: <AttachmentOutlinedIcon />,
+      component: <></>,
+    });
+  }
```

Comment:
Not sure if this will work without having the entity_id given in as well later which I'm not sure makes as much sense for a component called TabView so I would personally be inclined to let this part be duplicated outside the component. Not sure though - probably fine and will be clearer later when its used.

Reply:
Change gallery to galleryEntityId and attachments to attachmentsEntityId